runtime_config_linux.go

```diff
@@ -25,10 +25,10 @@ type LinuxRuntime struct {
 	// Resources contain cgroup information for handling resource constraints
 	// for the container
 	Resources *Resources `json:"resources,omitempty"`
-	// CgroupsPath specifies the path to cgroups that are created and/or joined by the container.
-	// The path is expected to be relative to the cgroups mountpoint.
-	// If resources are specified, the cgroups at CgroupsPath will be updated based on resources.
-	CgroupsPath *string `json:"cgroupsPath,omitempty"`
+	// CgroupsName specifies the name of to cgroups that are created and/or joined by the container.
+	// The name is expected to be absolute.
+	// If resources are specified, the CgroupsName cgroups will be updated based on resources.
+	CgroupsName string `json:"cgroupsName,omitempty"`
```

Comment:
On Mon, Dec 28, 2015 at 12:14:05PM -0800, Vish Kannan wrote:
> > +	// CgroupsName specifies the name of to cgroups that are created and/or joined by the container.
> > +	// The name is expected to be absolute.
> > +	// If resources are specified, the CgroupsName cgroups will be updated based on resources.
> > +	CgroupsName string `json:"cgroupsName,omitempty"`
> 
> Why? There are only two options here, an empty string "" or a valid name.
> There isn't a third system default.

This:

  "cgroupsName": "",

seems illegal to me (it's not absolute).  If we use missing/null
cgroupsName, we don't have to put in an absolute-path loophole for the
empty string.  For more background on nil over empty strings for
“unset”, see [1] and surrounding comments, as well as [2].

[1]: https://github.com/opencontainers/specs/pull/233#discussion_r47829711
[2]: https://github.com/opencontainers/specs/pull/270/files#r47838511